src/Viewer/services/decoder/FileManager.js

```diff
@@ -176,6 +190,37 @@ class FileManager {
         this.state.numberOfEvents = this._logEventOffsets.length;
     };
 
+    /**
+     * Given a ref timestamp, find the first log event whose timestamp is 
+     * greater or equal to the ref timestamp.
+     * Since all the log events are naturally sorted w.r.t. the timestamp,
+     * we can use binary search.
+     * Return the index of the log event.
+     * If the log event doesn't exist, the function should return null
+     */
+    _findFirstLogEventWithTimestamp (refTimestamp) {
+        const numberOfEvents = this._logEventOffsets.length;
+        let left = 0;
+        let right = numberOfEvents - 1;
+        let mid;
+
+        while (left <= right) {
+            mid = left + Math.floor((right - left) / 2); // To avoid overflow.
```

Comment:
* Could we write this as `mid = Math.floor((left + right) / 2);`?
* We typically use in-line comments for simple things like units. For a comment like yours, I would put it above the line and be explicit saying we use floor for x...

Reply:
Yes, we can. I was worried about integer overflow but seems like this is automatically handled by js already.

Reply:
Ah, gotcha. Yeah, I didn't get that from the comment. I would improve the comment and stick with your way then since I don't think JS handles overflow nicely, does it?